back/src/Guide/Api/Matomo.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+
+{- | Matomo curl requests -}
+module Guide.Api.Matomo
+       ( Matomo(..)
+       , postMatomo
+       ) where
+
+import Data.IP (IP)
+import Imports
+import Network.HTTP.Client (httpLbs, newManager, parseRequest, setQueryString)
+import Network.HTTP.Client.TLS (tlsManagerSettings)
+
+import Guide.Api.Guider (Guider)
+import Guide.Types.Edit (Edit (..))
+
+
+-- | Request with log data to send to Matomo
+data Matomo = Matomo
+  { mIP       :: Maybe IP
+  , mUA       :: Maybe Text
+  , mReferrer :: Maybe Text
+  , mTag      :: Edit
+  } deriving (Eq, Show)
+
+-- | Post log of any editions to Matomo.
+postMatomo :: Matomo -> Guider ()
+postMatomo Matomo{..} = liftIO $ do
+    manager  <- liftIO $ newManager tlsManagerSettings
+    req <- setQueryString
+      [ ("idsite", Just "1")
+      , ("rec", Just "1")
+      , ("url", toByteString . show <$> mIP)
+      , ("ua", toByteString <$> mUA)
+      , ("urlref", toByteString <$> mReferrer)
+      , ("action_name", Just . toByteString $ show mTag)
```

Comment:
We shouldn't send the whole `Edit` to Matomo, just the tag (i.e. `Edit'BlahBlah`, not `Edit'BlahBlah {foo = blah, bar = blah}`).